src/ledger/LedgerTxnContractDataSQL.cpp

```diff
@@ -161,9 +161,9 @@ class BulkLoadContractDataOperation
                           ") "
                           "SELECT ledgerentry "
                           "FROM contractdata "
-                          "WHERE contractid IN (SELECT value FROM r) "
-                          "AND key IN (SELECT value FROM r) "
-                          "AND type IN (SELECT value FROM r)";
+                          "WHERE contractid IN (SELECT contractid FROM r) "
```

Comment:
I'm curious - What was the symptom of this bug and how did you trigger it?

Reply:
This query would always return null. It gets called when you prefetch `ContractData` entries and are using SQLite without BucketListDB enabled. The result is `ContracttData` entries would never actually get prefetched and would be a guaranteed cache miss.

Reply:
The story here is interesting! I am going to chalk it up to "v5 of a PR that was a year and several hundred rebasings in the making"

- I think the code was actually originally written ok: https://github.com/stellar/stellar-core/blob/d4b12688c577110ec2c7b2ecb0303722cfba43cf/src/ledger/LedgerTxnContractDataSQL.cpp#L127-L131
- It remained ok until just the last Giant State Expiration PR Of Doom 2 weeks ago, where it was changed to this not-ok form here https://github.com/stellar/stellar-core/blob/d347617b3af1cae052411c883e44d8621676b295/src/ledger/LedgerTxnContractDataSQL.cpp#L153-L159
- It is a little funny since there's another edit in that PR just 40 lines later that does the ok thing: https://github.com/stellar/stellar-core/blob/d347617b3af1cae052411c883e44d8621676b295/src/ledger/LedgerTxnContractDataSQL.cpp#L190-L194
- In any event, none of us eagle-eyed reviewers caught it 2 weeks ago either, nor did our obviously inadequate tests, so .. here we are!

I would overall just prefer switching this back to the same idiom we used before and elsewhere in these clauses. I'll .. see if I can write it as a suggestion-comment using github's newfangled suggest-o-tron.